src/test/java/org/folio/eusage/reports/MainVerticleTest.java

```diff
@@ -1478,6 +1502,32 @@ public void testPostTenantOK(TestContext context) {
     }
     context.assertEquals(1, noPackages);
 
+    response = RestAssured.given()
+        .header(XOkapiHeaders.TENANT, tenant)
+        .header(XOkapiHeaders.URL, "http://localhost:" + MOCK_PORT)
+        .get("/eusage-reports/report-packages")
+        .then().statusCode(200)
+        .header("Content-Type", is("application/json"))
+        .extract();
+    resObject = new JsonObject(response.body().asString());
+    context.assertEquals(packageTitles.length, resObject.getJsonArray("packages").size());
+    for (int i = 0; i < packageTitles.length; i++) {
+      JsonObject kbPackage = resObject.getJsonArray("packages").getJsonObject(i);
+      context.assertEquals("good package name", kbPackage.getString("kbPackageName"));
+      context.assertEquals(goodPackageId.toString(), kbPackage.getString("kbPackageId"));
+      context.assertEquals(packageTitles[i].toString(), kbPackage.getString("kbTitleId"));
+    }
```

Comment:
Rest-assured is designed to be used with body(gpath, hamcrest matcher):

```suggestion
        .body("packages*.kbPackageName", everyItem(is("good package name")))
        .body("packages*.kbPackageId", everyItem(is(goodPackageId.toString())))
        .body("packages*.kbTitleId", containsInAnyOrder(asStrings(packageTitles)));
```

Note that containsInAnyOrder also checks that size so that we don't need `.body("packages", hasSize(packageTitles.length))` here.

Helper function asStrings:
```
  private static String [] asStrings(Object [] objects) {
    return Arrays.stream(objects).map(Object::toString).toArray(String[]::new);
  }
```

Reply:
The SQL query has no ORDER BY and may return the results in a different order. Using `containsInAnyOrder` takes care of this.